### ISSUE TITLE
DateTimeSerializer parses DateTime strings when they have exactly 2 decimal points

### ISSUE THREAD

**Issue by jeffjung-AI** [ServiceStack/Issues#324]
The following unit test will fail. I think they should return the same DateTime. 
When there are exactly 2 decimal places in the second, ```DateTimeSerializer.ParseShortestXsdDateTime()``` will return a local DateTime but with its ```Kind``` forced to be ```Utc```.
```csharp
[Test]
public void ParseShortestXsdDateTime_TwoDateTimesWithDifferentPrecision_ReturnsSameParsedDateTime()
{
    JsConfig.AssumeUtc = true;
    JsConfig.AlwaysUseUtc = true;

    const string noDecimalPoint = "1979-05-09T00:00:01Z";
    const string twoDecimalPoints = "1979-05-09T00:00:01.00Z";

    var dateTimeWithNoDecimalPoint = DateTimeSerializer.ParseShortestXsdDateTime(noDecimalPoint);
    var dateTimeWithTwoDecimalPoints = DateTimeSerializer.ParseShortestXsdDateTime(twoDecimalPoints);

    Assert.That(dateTimeWithNoDecimalPoint, Is.EqualTo(dateTimeWithTwoDecimalPoints));
}
```

The result is
```
Expected: 1979-05-08 17:00:01.000

But was:  1979-05-09 00:00:01.000
```

**Comment by mythz** [comment]
Should now be fixed [with this commit](https://github.com/ServiceStack/ServiceStack.Text/commit/acd96046a200febed28e9d8de8ad7d71d20056e9)

Change is available from v4.0.43+ that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet)

**Comment by jeffjung-AI** [comment]
Awesome! Thank you.
